hazelcast/src/main/java/com/hazelcast/datalink/impl/DataLinkConsistencyChecker.java

```diff
@@ -53,34 +52,39 @@ public void init() {
     }
 
     /**
-     * Removes & alters all outdated data links in __sql.catalog
-     * and adds all missed data links to __sql.catalog
+     * Removes & alters all outdated data links and adds
+     * all missed data links to data link service
      */
     public void check() {
         if (!initialized) {
             return;
         }
 
-        List<Map.Entry<String, DataLinkEntry>> sqlEntries = dataLinkService.getDataLinks()
-                .entrySet()
-                .stream()
-                .filter(en -> en.getValue().source == DataLinkSource.SQL)
-                .collect(Collectors.toList());
+        // capture data links set before altering it.
+        List<DataLink> sqlDataLinks = dataLinkService.getSqlCreatedDataLinks();
 
         for (Object catalogItem : sqlCatalog.values()) {
             if (!(catalogItem instanceof DataLinkCatalogEntry)) {
                 continue;
             }
             DataLinkCatalogEntry dl = (DataLinkCatalogEntry) catalogItem;
-            dataLinkService.replaceSqlDataLink(dl.getName(), dl.getType(), dl.getOptions());
+
+            // If a data link is found in the catalog that conflicts in name with one
+            // in the config, the one from the catalog should be deleted.
+            if (dataLinkService.existsConfigDataLink(dl.name())) {
```

Comment:
what is the use case for this? with this code in place you will not be able to move datalink from static config to SQL or vice-versa when doing rolling restart and using the link continuously.
this is also not a good way to handle inconsistent config of members.

Reply:
According to TDD
```
The data link has a unique name, it’s unique in config and in the SQL catalog.
If a data link is found in the catalog that conflicts in name with one in the
config, the one from the catalog is deleted by the background job. This happens
also after a data link is added through _dynamic config_.
```

Reply:
OK, we can keep it according to TDD, but I still think this can cause issues in practice.